lib/dash/segment_template.js

```diff
@@ -407,7 +422,17 @@ shaka.dash.SegmentTemplate = class {
 
     // If the availability timeline currently ends before the period, we will
     // need to add references over time.
-    if (presentationTimeline.getSegmentAvailabilityEnd() < periodEnd) {
+    const willNeedToAddReferences =
+        presentationTimeline.getSegmentAvailabilityEnd() < getPeriodEnd();
+
+    // When we start a live stream with a period that ends within the
+    // availability window we will not need to add more references, but we will
+    // need to evict old references.
+    const willNeedToEvictReferences = getPeriodEnd() != Infinity &&
+        (presentationTimeline.getSegmentAvailabilityStart() > periodStart) &&
+        (presentationTimeline.getSegmentAvailabilityEnd() > getPeriodEnd());
```

Comment:
I am not certain that this is the best way to compute that we will need to evict references.

Reply:
So you are right that there are two cases:

1. We need to add segments.  This happens when the period goes beyond the availability window.  Your logic for this already looks good to me.
2. We need to evict segments.  This happens when the period is available, but the availability duration is finite.  Try this, with the computation done in stages for clarity:

```js
const periodAvailable = availabilityStartTime <= getPeriodEnd();
const availabilityFinite = presentationTimeline.getSegmentAvailabilityDuration() != Infinity;
const willNeedToEvictReferences = periodAvailable && availabilityFinite;
```

The getter `getSegmentAvailabilityDuration()` will need to be added to `PresentationTimeline`.  There is an internal member for it already called `segmentAvailabilityDuration_`.

Reply:
will do

Reply:
@joeyparrish  what do you think about simplifying this to

`const willNeedToEvictReferences = presentationTimeline.isLive();
`

All live streams and only live streams have a segmentAvailabilityStart time that moves forward and therefore have segment references that that need to be evicted.

Reply:
That is broader, but some false positives should be okay.  A live stream can have an infinite availability duration (technically), in which case nothing would ever be evicted.  But it may not matter much if we ignore that case here.